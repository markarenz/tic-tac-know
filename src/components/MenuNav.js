import React from 'react';
import PropTypes from 'prop-types';

const MenuNav = ({ handleClickPrev, handleClickNext, mode }) => {
  return (
    <div className="grid grid-cols-2 mt-4 text-white">
      <div className="text-left">
        <button
          data-testid={`menu-nav-prev-${mode}`}
          onClick={handleClickPrev}
          className="text-4xl hover:scale-125 transition-scale duration-300"
        >
          &#9664;
        </button>
      </div>
      <div className="text-right">
        <button
          data-testid={`menu-nav-next-${mode}`}
          onClick={handleClickNext}
          className="text-4xl hover:scale-125 transition-scale duration-300"
        >
          &#9654;
        </button>
      </div>
    </div>
  );
};

MenuNav.propTypes = {
  handleClickPrev: PropTypes.func.isRequired,
  handleClickNext: PropTypes.func.isRequired,
  mode: PropTypes.string.isRequired,
};
export default MenuNav;
