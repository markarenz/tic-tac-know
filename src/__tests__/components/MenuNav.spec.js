import { render, fireEvent, screen } from '@testing-library/react';
import MenuNav from '../../components/MenuNav';

describe('Thinking', () => {
  const mocks = {
    handleClickPrev: jest.fn(),
    handleClickNext: jest.fn(),
  };
  it('renders component', () => {
    const container = render(
      <MenuNav handleClickPrev={mocks.handleClickPrev} handleClickNext={mocks.handleClickNext} />
    );
    expect(container).toMatchSnapshot();
  });
  it('triggers handleClickPrev on click', () => {
    const container = render(
      <MenuNav handleClickPrev={mocks.handleClickPrev} handleClickNext={mocks.handleClickNext} />
    );
    fireEvent(
      screen.getByTestId('menu-nav-prev'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );
    expect(mocks.handleClickPrev).toHaveBeenCalled();
  });
  it('triggers handleClickNext on click', () => {
    const container = render(
      <MenuNav handleClickPrev={mocks.handleClickPrev} handleClickNext={mocks.handleClickNext} />
    );
    fireEvent(
      screen.getByTestId('menu-nav-next'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );
    expect(mocks.handleClickNext).toHaveBeenCalled();
  });
  it('matches snapshot', () => {
    const container = render(
      <MenuNav handleClickPrev={mocks.handleClickPrev} handleClickNext={mocks.handleClickNext} />
    );
    expect(container).toMatchSnapshot();
  });
});
