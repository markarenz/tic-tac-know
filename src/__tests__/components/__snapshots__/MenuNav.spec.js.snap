// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thinking matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid-cols-2 mt-4 text-white"
      >
        <div
          class="text-left"
        >
          <button
            class="text-4xl hover:scale-125 transition-scale duration-300"
            data-testid="menu-nav-prev-test"
          >
            ◀
          </button>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-4xl hover:scale-125 transition-scale duration-300"
            data-testid="menu-nav-next-test"
          >
            ▶
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid-cols-2 mt-4 text-white"
    >
      <div
        class="text-left"
      >
        <button
          class="text-4xl hover:scale-125 transition-scale duration-300"
          data-testid="menu-nav-prev-test"
        >
          ◀
        </button>
      </div>
      <div
        class="text-right"
      >
        <button
          class="text-4xl hover:scale-125 transition-scale duration-300"
          data-testid="menu-nav-next-test"
        >
          ▶
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Thinking renders component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid-cols-2 mt-4 text-white"
      >
        <div
          class="text-left"
        >
          <button
            class="text-4xl hover:scale-125 transition-scale duration-300"
            data-testid="menu-nav-prev-test"
          >
            ◀
          </button>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-4xl hover:scale-125 transition-scale duration-300"
            data-testid="menu-nav-next-test"
          >
            ▶
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid-cols-2 mt-4 text-white"
    >
      <div
        class="text-left"
      >
        <button
          class="text-4xl hover:scale-125 transition-scale duration-300"
          data-testid="menu-nav-prev-test"
        >
          ◀
        </button>
      </div>
      <div
        class="text-right"
      >
        <button
          class="text-4xl hover:scale-125 transition-scale duration-300"
          data-testid="menu-nav-next-test"
        >
          ▶
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
